src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -159,51 +204,106 @@ public static List<int> CommonBaudRates()
             };
         }
 
+        /// <summary>
+        /// Adds a new command handler.
+        /// A command handler can support extended commands.
+        /// </summary>
+        /// <typeparam name="T">An instance of <see cref="ExtendedCommandHandler"/>.</typeparam>
+        /// <param name="newCommandHandler">The new handler</param>
+        public void AddCommandHandler<T>(T newCommandHandler)
+            where T : ExtendedCommandHandler
+        {
+            if (newCommandHandler == null)
+            {
+                throw new ArgumentNullException(nameof(newCommandHandler));
+            }
+
+            if (newCommandHandler.HandlesMode != null)
+            {
+                lock (_knownSupportedModes)
+                {
+                    // If we already know the mode, replace its configuration with the new one (typically, this will only update the name)
+                    var m = _knownSupportedModes.FirstOrDefault(x => x.Value == newCommandHandler.HandlesMode.Value);
+                    if (m != null)
+                    {
+                        _knownSupportedModes.Remove(m);
+                    }
+
+                    _knownSupportedModes.Add(newCommandHandler.HandlesMode);
+                }
+            }
+
+            _extendedCommandHandlers.Add(newCommandHandler);
+            if (_firmata != null)
+            {
+                // Only if already initialized
+                newCommandHandler.Registered(_firmata, this);
+            }
+        }
+
+        /// <summary>
+        /// Gets the command handler with the provided type.
+        /// </summary>
+        /// <typeparam name="T">The type to query</typeparam>
+        /// <returns>The command handler, or null if none was found</returns>
+        public T? GetCommandHandler<T>()
+            where T : ExtendedCommandHandler
+        {
+            foreach (var cmd in _extendedCommandHandlers)
+            {
+                if (cmd.GetType() == typeof(T))
+                {
+                    return (T)cmd;
+                }
+            }
+
+            throw new NotSupportedException($"No handler of type {typeof(T)} found");
+        }
+
+        /// <summary>
+        /// Gets the command handler with the provided type.
+        /// </summary>
+        /// <typeparam name="T">The type to query</typeparam>
+        /// <param name="handler">The out variable</param>
+        /// <returns>True on success, false otherwise</returns>
+        public bool TryGetCommandHandler<T>(
+#if NET5_0_OR_GREATER
```

Comment:
we should consider doing something similar the dotnet/runtime does instead (not in this PR) - they add dummy types related to nullability during the build so that it compiles when we're not on .NET 5 as well. This makes it much easier as you don't need to `#if` everywhere